### ISSUE TITLE
Allow additional packages to be added to shells

### ISSUE THREAD

**Issue by 3noch** [obsidiansystems/obelisk#189]
Right now `shells.ghc` is hard-coded to include only `backend`, `frontend`, and `common`. It would be very handy if we could add others.

**Comment by ryantrinkle** [comment]
Would you mind sharing a specific use case?  This sounds reasonable, but I'd like to understand in some more detail.

**Comment by 3noch** [comment]
In our case we have split off a large number of the types into a different cabal project that can be shared on its own. While it's easy to use this library with obelisk (by just adding it to `projects`), active development is painful since any change to the library requires a restart of `ob run` and a full rebuild of the library. I've enjoyed having this feature in reflex-platform's more generic `project` system, and now I'm missing it in an obelisk project.

**Comment by ryantrinkle** [comment]
Hmm, if the library is a (transitive) dependency of backend, frontend, or common, it and all its deps should already be included.  What have you tried so far to get this to work?

**Comment by 3noch** [comment]
It's a separate cabal project (and associated folder structure), so not mentioning it at all in `packages` or `overrides` causes nix to fail because it can't open the shell without a required argument (dependency declared in cabal file). If I add it to either of those places, it builds it via nix before opening the shell. I've also tried adding it to `cabal.project` but `ob run`'s `ghcid` invocation doesn't rely on cabal projects AFAIK.

**Comment by 3noch** [comment]
To be clear, the problem is that I want to make this extra project *not* a nix dependency of the shell, as is the case of `backend`, `frontend`, and `common` (but still include *it's* dependencies). This is what reflex-platform's project shells let you do.

**Comment by 3noch** [comment]
The other part of this, for obelisk, would be ensuring that `ghcid` pulls in the right files for `ob run`.

**Comment by 3noch** [comment]
Cf. https://github.com/obsidiansystems/obelisk/blob/8dca44efd8b8b5d638b178c1be4ced9b3e3fd16c/lib/command/src/Obelisk/Command/Run.hs#L51

**Comment by mightybyte** [comment]
I would also like this.  The immediate situation I just encountered is that some GHCJS dependencies don't build with the newest version of a dependency that I'm trying out.  I'm only doing backend stuff at the moment and don't want to deal with that dependency problem right now.  But more importantly, I've started using this infrastructure for more generic multi-package projects and I'd like this power.

I would also like if you would bring back the ability to name the project so I have more descriptive names than `reflex-project`.

**Comment by mulderr** [comment]
This is probably a bit obscure but, I think, still relevant use case.

I am in the process of upgrading project "foo" to use obelisk. It includes a foo-db package for [beam](http://hackage.haskell.org/package/beam) queries and various related stuff - just as a way of organizing code. foo-db depends on common because that is where the schema is defined.

Where `nix-build -A ghc.backend` works fine `ob run` reports lots of these kind of errors in a backend module that uses both common and foo-db:
```
backend/src/Foo/Api.hs:313:66: error:
    • Couldn't match type ‘common-0.1:Common.Schema.Foo.AttachmentT
                             Identity’
                     with ‘AttachmentT Identity’
      NB: ‘AttachmentT’ is defined at
            common/src/Common/Schema/Foo.hs:(29,1)-(41,22)
          ‘common-0.1:Common.Schema.Foo.AttachmentT’
            is defined in ‘Common.Schema.Foo’ in package ‘common-0.1’
```
What is going on is that one version of common comes from nix as a dependency of foo-db and a completely different version is loaded by ghci directly from source. As already mentioned the ghci script generated by `ob run` loads a hard coded list of packages so this is not fixable unless we reorganize things - ex. merge foo-db into backend, or alternatively, split the schema definion into yet another package.

So in this case nix knowing about the additional package while obelisk does not causes inconsistent behavior between the two.

--

Can we store the list of packages/shells in a separate configuration file that would be used both by nix and obelisk?

**Comment by 3noch** [comment]
This was resolved as of v0.5: https://github.com/obsidiansystems/obelisk/releases/tag/v0.5.0.0
